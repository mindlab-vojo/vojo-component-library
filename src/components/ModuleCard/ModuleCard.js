import React from 'react'
import PropTypes from 'prop-types'

import Card from '../Card'
import IconManager from '../IconManager'
import Typography from '../Typography'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import * as ModuleCard from './ModuleCard.style'

const ModuleCardComponent = ({
  description,
  iconName,
  iconBackgroundColor,
  lineHeight,
  onClickCard,
  title,
  time,
}) => {

  const renderComponent = (theme) => (
    <Card
      hasPointer
      onClickCard={onClickCard}>
      <ModuleCard.Wrapper
        className="ModuleCard">
        <ModuleCard.IconWrapper
          className="ModuleCard__IconWrapper">
          <ModuleCard.IconCircle
            backgroundColor={iconBackgroundColor}
            className="ModuleCard__IconCircle">
            <IconManager
              height="24px"
              fill={theme.colors.whiteColor}
              icon={iconName}
              width="24px"
            />
          </ModuleCard.IconCircle>
        </ModuleCard.IconWrapper>
        <ModuleCard.InfoWrapper
          lineHeight={lineHeight}
          className="ModuleCard__InfoWrapper">
          <ModuleCard.Title
            className="ModuleCard__InfoWrapper__Title">
            <Typography
              color={theme.colors.darkBlueColor}
              fontSize="18px"
              fontWeight="bolder"
              fontFamily="'Montserrat', sans-serif">
              {title}
            </Typography>
          </ModuleCard.Title>
          <ModuleCard.Description
            className="ModuleCard__InfoWrapper__Description">
            <Typography
              fontSize="12px">
              {description}
            </Typography>
          </ModuleCard.Description>
          <ModuleCard.Time
            className="ModuleCard__InfoWrapper__Time">
            <Typography
              fontSize="12px"
              fontWeight="bolder">
              {time}
            </Typography>
          </ModuleCard.Time>
        </ModuleCard.InfoWrapper>
      </ModuleCard.Wrapper>
    </Card>
  )
  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

ModuleCardComponent.propTypes = {
  description: PropTypes.string,
  iconName: PropTypes.string,
  iconBackgroundColor: PropTypes.string,
  lineHeight: PropTypes.string,
  imageUrl: PropTypes.string,
  onClickCard: PropTypes.func,
  title: PropTypes.string,
  time: PropTypes.string,
}

export default ModuleCardComponent
