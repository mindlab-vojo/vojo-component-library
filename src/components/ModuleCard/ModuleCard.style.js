import styled from 'styled-components'

export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  width: 100%;
`

export const IconWrapper = styled.div`
  margin: 0 15px 0 10px;
`

export const IconCircle = styled.div`
  align-items: center;
  background-color: ${props => props.backgroundColor};
  border-radius: 120px;
  display: flex;
  height: 38px;
  justify-content: center;
  width: 38px;
`

export const InfoWrapper = styled.div``

export const Title = styled.div``

export const Description = styled.div``

export const Time = styled.div``