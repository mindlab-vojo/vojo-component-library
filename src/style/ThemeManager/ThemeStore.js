import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore