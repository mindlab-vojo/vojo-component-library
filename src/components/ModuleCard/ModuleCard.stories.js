import React from 'react';
import { withKnobs, text, color, select } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import ModuleCard from '.'; 
import { iconList } from '../IconManager/IconManager'
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const moduleCard = () => {
  const description = text("description", "Formulário com informações básicas sobre você para a função que deseja")
  const iconName = select('iconName', Object.keys(iconList), Object.keys(iconList)[0])
  const iconBackgroundColor = color("iconBackgroundColor", "#FF5400")
  const lineHeight = text('lineHeight', '1.1')
  const onClickCard = action("onClickCard")
  const title = text("title", "Formulário de Inscrição")
  const time = text("time", "5min.")
  const titleSize = text("titleSize", "18px")
  const iconCircleSize = text("iconCircleSize", "38px")

  return (
    <ModuleCard
      description={description}
      iconName={iconName}
      iconBackgroundColor={iconBackgroundColor}
      lineHeight={lineHeight}
      onClickCard={onClickCard}
      title={title}
      time={time}
      titleSize={titleSize}
      iconCircleSize={iconCircleSize}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'ModuleCard',
  component: moduleCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  