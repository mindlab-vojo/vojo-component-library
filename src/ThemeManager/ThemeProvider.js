import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer