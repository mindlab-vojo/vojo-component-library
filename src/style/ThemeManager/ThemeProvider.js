import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer