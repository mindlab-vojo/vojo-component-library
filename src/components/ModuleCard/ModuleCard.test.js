import React from 'react';
import { render } from '@testing-library/react';

import ModuleCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test ModuleCard component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <ModuleCard 
        description="Just testing by"
        iconName="Check"
        iconBackgroundColor="#DF74A3"
        onClickCard={fakeFn}
        title="Just a Title"
        time="50min."/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
