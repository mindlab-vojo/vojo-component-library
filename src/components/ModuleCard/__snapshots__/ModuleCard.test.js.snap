// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ModuleCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm hCpYkD Card"
      >
        <div
          class="sc-AxirZ iTEWhx ModuleCard"
        >
          <div
            class="sc-AxiKw kDJXyU ModuleCard__IconWrapper"
          >
            <div
              class="sc-AxhCb iFtFSU ModuleCard__IconCircle"
            >
              <svg
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                  fill="#FFFFFF"
                />
              </svg>
            </div>
          </div>
          <div
            class="sc-AxhUy cjNzFp ModuleCard__InfoWrapper"
          >
            <div
              class="sc-AxgMl fFPQpl ModuleCard__InfoWrapper__Title"
            >
              <p
                class="sc-AxmLO euTaYe"
              >
                Just a Title
              </p>
            </div>
            <div
              class="sc-AxheI fniENP ModuleCard__InfoWrapper__Description"
            >
              <p
                class="sc-fzozJi Wya-Dg"
              >
                Just testing by
              </p>
            </div>
            <div
              class="sc-Axmtr hBVTto ModuleCard__InfoWrapper__Time"
            >
              <p
                class="sc-fzoLsD jeWUtM"
              >
                50min.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm hCpYkD Card"
    >
      <div
        class="sc-AxirZ iTEWhx ModuleCard"
      >
        <div
          class="sc-AxiKw kDJXyU ModuleCard__IconWrapper"
        >
          <div
            class="sc-AxhCb iFtFSU ModuleCard__IconCircle"
          >
            <svg
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                fill="#FFFFFF"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-AxhUy cjNzFp ModuleCard__InfoWrapper"
        >
          <div
            class="sc-AxgMl fFPQpl ModuleCard__InfoWrapper__Title"
          >
            <p
              class="sc-AxmLO euTaYe"
            >
              Just a Title
            </p>
          </div>
          <div
            class="sc-AxheI fniENP ModuleCard__InfoWrapper__Description"
          >
            <p
              class="sc-fzozJi Wya-Dg"
            >
              Just testing by
            </p>
          </div>
          <div
            class="sc-Axmtr hBVTto ModuleCard__InfoWrapper__Time"
          >
            <p
              class="sc-fzoLsD jeWUtM"
            >
              50min.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
